tools/devops/automation/templates/common/setup.yml

```diff
@@ -67,7 +67,12 @@ steps:
   continueOnError: true
 
 - bash: |
-    security set-key-partition-list -S apple-tool:,apple: -s -k $OSX_KEYCHAIN_PASS login.keychain
+    security set-key-partition-list -S apple-tool:,apple: -s -k $OSX_KEYCHAIN_PASS login.keychain || RC=$?
+    if [ $RC -eq 0 ]; then
```

Comment:
RC will never be 0, because `RC=$?` won't be executed in that case.

You need to do `RC=0` first.